code/mob/living/critter/ai/flock/flocktasks.dm

```diff
@@ -586,22 +585,19 @@ butcher
 		var/mob/living/critter/flock/drone/F = holder.owner
 		usr = F // don't ask, please, don't
 		if(F?.set_hand(1)) // grip tool
-			if(!F.is_in_hands(container_target)) //if it's not in any of our hands, pick it up
-				F.drop_item() // drop whatever we're holding
+			if(!F.is_in_hands(container_target))
+				F.drop_item()
 				F.set_dir(get_dir(F, container_target))
 				F.hand_attack(container_target) //try and pick it up
-			if(F.is_in_hands(container_target)) //did we do it?
```

Comment:
useful comment

Reply:
Idk, seems like is_in_hands() kind of already explains this except for intent, maybe would be better to just clarify that this won't work for items that have HUD's open or something?

Reply:
Honestly this whole block is a nightmare, so I just wanted to explain at every step what is happening